execution/block-partitioner/src/v2/conflicting_txn_tracker.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright © Aptos Foundation
+
+use crate::v2::{OriginalTxnIdx, ShardedTxnIndex2};
+#[cfg(test)]
+use aptos_types::state_store::state_key::StateKey;
+use aptos_types::{
+    block_executor::partitioner::{RoundId, ShardId},
+    transaction::analyzed_transaction::StorageLocation,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::{btree_set::BTreeSet, HashSet};
+
+/// This structure is only used in `V2Partitioner`.
+/// For txns that claimed to access the same storage location,
+/// it caches some metadata about the location and also keeps track of their status (pending or position finalized) throughout the partitioning process.
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct ConflictingTxnTracker {
+    /// The storage location on which conflicting txns are being tracked by this tracker.
+    pub storage_location: StorageLocation,
+    /// A randomly chosen owner shard of the storage location, for conflict resolution purpose.
+    pub anchor_shard_id: ShardId,
+    /// Txns that (1) read the current storage location and (2) have not been accepted.
+    pending_reads: BTreeSet<OriginalTxnIdx>,
+    /// Txns that (1) write the current storage location and (2) have not been accepted.
+    pending_writes: BTreeSet<OriginalTxnIdx>,
+    /// Txns that write the current storage location.
+    pub writer_set: HashSet<OriginalTxnIdx>,
+    /// Txns that have been accepted.
+    pub finalized_all: BTreeSet<ShardedTxnIndex2>,
+    /// Txns that (1) write the current storage location and (2) have been accepted.
+    pub finalized_writes: BTreeSet<ShardedTxnIndex2>,
+}
+
+impl ConflictingTxnTracker {
+    pub fn new(storage_location: StorageLocation, anchor_shard_id: ShardId) -> Self {
```

Comment:
nit: rather than passing the `anchor_shard_id`, you can just calculate here. That would make the constructor cleaner.

Reply:
then the constructor needs a `num_shards`... Not much difference?

Reply:
IMO, its just cleaner to pass `num_shards` in the constructor.. but up to you.